add balanced batching to the docs (#4348)
RasaHQ/rasa: docs/nlu/choosing-a-pipeline.rst

    You need to provide enough data for both intents and entities.

Class imbalance
---------------

> In order to mitigate class imbalance problem, we introduced ``balanced`` batching strategy for

Reviewer: ```suggestion
Classification algorithms often do not perform well if there is a large `class imbalance`, for example if you have a lot of training data for some intents and very little training data for others. 
To mitigate this problem, rasa's ``supervised_embeddings`` pipeline uses a ``balanced`` batching strategy.
```

> ``supervised_embeddings`` pipeline. This algorithm ensures that all classes are represented

Reviewer: ```suggestion
This algorithm ensures that all classes are represented
```

Author: but it is only for embed intent classifier, it doesn't work for space one

> at least in subsequent batches, still mimicking the fact that some classes are more frequent than others.

Reviewer: do you mean at least once _every other_ batch?

Author: that depends on the number of classes and batch size it could be once in every 3, but importantly there is a sequence of batches that will have all classes

Reviewer: ok. that's quite complicated to explain. we could say  

"ensures that all classes are represented in every batch, or at least in every `n`th batch (with `n` as small as possible)"

Author: do you think it simpler?

Author: then we'd need to explain what n is

Reviewer: what about _in every batch, or at least in as many batches as possible_

Author: As many subsequent batches as possible?

Reviewer: sounds good

> It is turned on by default. In order to turn it off and use classic batching strategy pass

Reviewer: ```suggestion
Balanced batching is used by default. In order to turn it off and use a classic batching strategy include
```

> ``batch_strategy: sequence`` in config file.

Reviewer: ```suggestion
``batch_strategy: sequence`` in your config file.
```
